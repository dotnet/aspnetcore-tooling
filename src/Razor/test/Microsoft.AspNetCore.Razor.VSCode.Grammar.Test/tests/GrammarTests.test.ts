/* --------------------------------------------------------------------------------------------
* Copyright (c) Microsoft Corporation. All rights reserved.
* Licensed under the MIT License. See License.txt in the project root for license information.
* ------------------------------------------------------------------------------------------ */

import { RunAddTagHelperDirectiveSuite } from './AddTagHelperDirective';
import { RunCodeBlockSuite } from './CodeBlock';
import { RunCodeDirectiveSuite } from './CodeDirective';
import { RunExplicitExpressionSuite } from './ExplicitExpressions';
import { RunFunctionsDirectiveSuite } from './FunctionsDirective';
import { RunImplicitExpressionSuite } from './ImplicitExpressions';
import { RunModelDirectiveSuite } from './ModelDirective';
import { RunPageDirectiveSuite } from './PageDirective';
import { RunRemoveTagHelperDirectiveSuite } from './RemoveTagHelperDirective';
import { RunTagHelperPrefixDirectiveSuite } from './TagHelperPrefixDirective';
import { RunTransitionsSuite } from './Transitions';

// We bring together all test suites and wrap them in one here. The reason behind this is that
// modules get reloaded per test suite and the vscode-textmate library doesn't support the way
// that Jest reloads those modules. By wrapping all suites in one we can guaruntee that the
// modules don't get torn down inbetween suites.

describe('Grammar tests', () => {
    RunTransitionsSuite();
    RunExplicitExpressionSuite();
    RunImplicitExpressionSuite();
    RunCodeBlockSuite();

    // Directives
    RunCodeDirectiveSuite();
    RunFunctionsDirectiveSuite();
    RunPageDirectiveSuite();
    RunAddTagHelperDirectiveSuite();
    RunRemoveTagHelperDirectiveSuite();
    RunTagHelperPrefixDirectiveSuite();
    RunModelDirectiveSuite();
});
