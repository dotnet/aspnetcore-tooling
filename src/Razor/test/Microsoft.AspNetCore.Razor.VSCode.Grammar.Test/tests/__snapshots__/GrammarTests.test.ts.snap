// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grammar tests Transitions Escaped transitions 1`] = `
"Line: @@
 - token from 0 to 2 (@@) with scopes text.aspnetcorerazor, constant.character.escape.razor.transition
"
`;
